### PR TITLE
Editorial: Improve consistency of referencing Unicode documents

### PR DESCRIPTION
**Pull request by gibson042**
Excludes lines that are already being addressed by #735.

### CODE REVIEW

File: spec/segmenter.html [hunk 1]
```diff
@@ -391,7 +391,7 @@ <h1>Abstract Operations for Segmenter Objects</h1>
...
-      <emu-note>Boundary determination is implementation-dependent, but general default algorithms are specified in Unicode Standard Annex 29 (available at <a href="https://www.unicode.org/reports/tr29/">https://www.unicode.org/reports/tr29/</a>). It is recommended that implementations use locale-sensitive tailorings such as those provided by the Common Locale Data Repository (available at <a href="http://cldr.unicode.org">http://cldr.unicode.org</a>).</emu-note>
+      <emu-note>Boundary determination is implementation-dependent, but general default algorithms are specified in <a href="https://unicode.org/reports/tr29/">Unicode Standard Annex #29</a>. It is recommended that implementations use locale-sensitive tailorings such as those provided by the Common Locale Data Repository (available at <a href="http://cldr.unicode.org">http://cldr.unicode.org</a>).</emu-note>
```
Comment (ryzokuken) [inline spec/segmenter.html:394]: Can we change the CLDR link too?

**Review by ryzokuken — APPROVED** [review]
Overall looks great to me

File: spec/segmenter.html (hunk 1 above)
Comment (srl295) [inline spec/segmenter.html:394]:
```suggestion
      <emu-note>Boundary determination is implementation-dependent, but general default algorithms are specified in <a href="https://unicode.org/reports/tr29/">Unicode Standard Annex #29</a>. It is recommended that implementations use locale-sensitive tailorings such as those provided by the Common Locale Data Repository (available at <a href="https://cldr.unicode.org">https://cldr.unicode.org</a>).</emu-note>
```

**Review by srl295 — APPROVED** [review]
Lgtm with a suggested improvement. 

Note I'm proposing to drop the section numbers from uts 35 as they add a lot of editorial burden. https://unicode-org.atlassian.net/browse/CLDR-15473 and I noticed you filed a ticket related to numbering.

**Comment by gibson042** [comment]
> Note I'm proposing to drop the section numbers from uts 35 as they add a lot of editorial burden. https://unicode-org.atlassian.net/browse/CLDR-15473 and I noticed you filed a ticket related to numbering.

Excellent, I look forward to a future update further simplifying things here!

**Comment by srl295** [comment]
> > Note I'm proposing to drop the section numbers from uts 35 as they add a lot of editorial burden. https://unicode-org.atlassian.net/browse/CLDR-15473 and I noticed you filed a ticket related to numbering.
> 
> 
> 
> Excellent, I look forward to a future update further simplifying things here!

Me too

**Comment by srl295** [comment]
> > Note I'm proposing to drop the section numbers from uts 35 as they add a lot of editorial burden. https://unicode-org.atlassian.net/browse/CLDR-15473 and I noticed you filed a ticket related to numbering.
> 
> 
> 
> Excellent, I look forward to a future update further simplifying things here!

Me too

**Pull request merged**